api/Controllers/userController.ts

```diff
@@ -56,10 +57,10 @@ const profileDataInsertion = (userType: string) => {
 
 export const register = async (req: Request, res: Response) => {
   const {
-    body: { email, userType, password },
+    body: { email, userName, userType, password },
   } = req;
   try {
-    const userObject: InterfaceUserModel | null = await User.findOne({ email });
+    const userObject: InterfaceUserModel | null = await User.findOne({ $or: [ { email }, { userName } ] });
```

Comment:
do you really think we need or operator here, as described in schema `username` is `unique` so DB will throw an error if `username/ email` is not unique? and I think in your case it won't even go to userName, as email will never be null. Let me know your thought on it, or correct me if I am wrong.

Reply:
Ok so I have put an or condition for a case like this. Let's assume we already have a user { ashu, ashu@gmail.com} . This particular condition would prevent a user { ashu, ashuK@gmail.com} , {ashuK, ashu@gmail.com} and { ashu, ashu@gmail.com} from registering. This condition ensures uniqueness of both email and username for a new user. In login controller, this same condition ensures the user can either login with user name or email.

Reply:
I just modified the query. I think we need the userObject for socialLoginCheck function.

Reply:
so as far as I know or works like that if any condition is true it will go in, so if `{ ashu, ashu@gmail.com}` is registered, and the user is trying to register `{ ashuK, ashuK@gmail.com}` then here `ashuK` is unique so the condition will evaluate to true isn't work that way?

Reply:
Yeah looks good, didn't know about `$or` operator, though my initiation was defining `unique` in schema ensures `uniqueness` of both email and username for a new user.